### ISSUE TITLE
ENH: Allow to skip series type check in assert_frame_equal

### ISSUE THREAD

**Issue by abalkin** [pandas-dev/pandas#48287]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

When using derived Series type in dataframes, it is useful to be able to skip series type check in assert_frame_equal.

```python
import pandas as pd

class Series(pd.Series):
    @property
    def _constructor( self ):
        return Series

    @property
    def _constructor_expanddim( self ):
        return DataFrame

class DataFrame(pd.DataFrame):
    @property
    def _constructor_sliced( self ):
        return Series

s1 = pd.Series( [0] )
s2 = Series( s1 )

pd.testing.assert_series_equal(s1, s2, check_series_type=False)

df1 = s1.to_frame()
df2 = s2.to_frame()

pd.testing.assert_frame_equal(df1, df2, check_frame_type=False)
```

In the code above, the first assert passes, but the second fails with the following message:

```text
DataFrame.iloc[:, 0] (column name="0") classes are different
[left]:  Series
[right]: Series
```

There is currently no easy way to make such dataframes compare equal in testing.

### Feature Description

I suggest the following enhancements:

1. Add a new parameter, `check_series_type` to `assert_frame_equal()` that will pass through to `assert_series_equal()`.
1. Set  the new `check_series_type` to `False` when `check_frame_type` is set to `False`.
1. When type mismatch is detected but type names are the same, include fully qualified names in the assert message. 

### Alternative Solutions

Implement enhancements 1 and 3 without 2.

**Comment by alexander-beedie** [comment]
Like it; very useful.

**Comment by lucaslarios** [comment]
take it

**Comment by lucaslarios** [comment]
take

**Comment by lucaslarios** [comment]
Does this issue still remain relevant?

**Comment by alexander-beedie** [comment]
Yes. Seems somebody closed the proposed PR (which works and fixes the issue) as "stale" - but it's still valid.

**Comment by jreback** [comment]
@alexander-beedie and your anyone else
could pick up the proposed PR

things are stale if they r just sitting around for too long
